src/os/source/vectorsource.js

```diff
@@ -987,6 +987,55 @@ os.source.Vector.prototype.setColumnAutoDetectLimit = function(value) {
 };
 
 
+/**
+ * Sets the colors of the provided features
+ *
+ * @param {Array<!ol.Feature>|null} items
+ * @param {string|null} color
+ *
+ * @suppress {accessControls}
+ */
+os.source.Vector.prototype.setColor = function(items, color) {
+  if (!items || !color) return;
+
+  var cm = this.colorModel;
+  if (cm) {
+    cm.colorFeatures(items, color);
+  } else {
+    // TODO figure out how to do the binning when there are no bins...
+    for (const item of items) {
+      var oldColor = os.color.toHexString(os.feature.getColor(item, undefined, undefined));
+      if (color != oldColor) {
+        item.values_[os.data.RecordField.COLOR] = color;
+        item.dispatchFeatureEvent(os.data.FeatureEventType.COLOR, color, oldColor);
+      }
+    }
+  }
+  this.dispatchEvent(new os.events.PropertyChangeEvent(os.source.PropertyChange.STYLE));
+  this.changed();
+};
+
+
+/**
+ * Finds the first color in the items so the User has some consistency
+ *
+ * @param {Array<!ol.Feature>|null} items
+ * @return {string} the color
+ */
+os.source.Vector.prototype.getFirstColor = function(items) {
```

Comment:
This function seems better suited as `os.feature.getFirstColor`, as it doesn't relate to the source at all.